ocs_ci/framework/pytest_customization/marks.py

```diff
@@ -342,6 +368,27 @@
     reason="Test will not run on Managed service with provider and consumer clusters",
 )
 
+skipif_hci_provider = pytest.mark.skipif(
+    config.default_cluster_ctx.ENV_DATA["platform"].lower()
+    in HCI_PROVIDER_CLIENT_PLATFORMS
+    and config.default_cluster_ctx.ENV_DATA["cluster_type"].lower() == "provider",
+    reason="Test will not run on Fusion HCI provider cluster",
+)
+
+skipif_hci_client = pytest.mark.skipif(
+    config.default_cluster_ctx.ENV_DATA["platform"].lower()
+    in HCI_PROVIDER_CLIENT_PLATFORMS
+    and config.default_cluster_ctx.ENV_DATA["cluster_type"].lower() == "hci_client",
+    reason="Test will not run on Fusion HCI client cluster",
+)
+
+skipif_hci_provider_and_client = pytest.mark.skipif(
+    config.ENV_DATA["platform"].lower() in HCI_PROVIDER_CLIENT_PLATFORMS
+    and config.is_hci_provider_exist()
+    and config.is_hci_client_exist(),
```

Comment:
So the tests marked with this mark will not be skipped if run on hci provider without clients. Is it the desired behaviour? Wouldn't it be more useful to have a mark for skipping the test on any hci cluster, no matter if it's a provider, client or provider+client multicluster setup?

Reply:
This is similar to MS function https://github.com/red-hat-storage/ocs-ci/blob/b16f17fbc17c8e774b0fec587fe4a07c3659d191/ocs_ci/framework/pytest_customization/marks.py#L338

this function is also reference from MS Markers

Reply:
Also as now platform type is unique to hci , we can do this in either way